### PR TITLE
Fix first publish analytics

### PR DESCRIPTION
**Pull request by mzorz**
This PR fixes a problem in Tracks first spot by @elibud 

**tl;dr:** when you tap PUBLISH, we were setting the status of the Post to `published` immediately and checking _only after_ whether it was a first time publish. Thus, the check returned `false` because it wasn't a DRAFT anymore.

In previous code, when you tap on PUBLISH, the Post's status was being changed immediately to `PUBLISHED` and then being sent to the UploadService, with the problem that checks being performed for Tracks implied that the Post was not a DRAFT being published anymore, but rather looked like an update of an already published Post.
So this PR changes things a bit and only sets the Posts' new status  (PUBLISHED) after making the `isFirstTimePublished()` checks which we need to correctly track fresh / new posts.

To test:
1. start a draft
2. type something (Title/body)
3. tap on PUBLISH on the navigation bar
4. if you put a breakpoint , observe the Tracks code being called is `UploadService.uploadPostAndTrackAnalytics` in https://github.com/wordpress-mobile/WordPress-Android/compare/release/12.2...try/fix-first-publish-analytics?expand=1#diff-38b6aea9cfff5f3e0e4b7389b44de602R1895

Update release notes:

- [x] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java [hunk 1]
```diff
@@ -1741,9 +1741,7 @@ public void onPositiveClicked(@NonNull String instanceTag) {
                 mZendeskHelper.createNewTicket(this, Origin.DISCARD_CHANGES, mSite);
                 break;
             case TAG_PUBLISH_CONFIRMATION_DIALOG:
-                mPost.setStatus(PostStatus.PUBLISHED.toString());
-                mPostEditorAnalyticsSession.setOutcome(Outcome.PUBLISH);
```
Comment (mzorz) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: Setting the status of the Post immediately made the `isFirstTimePublish()` check to fail so, we moved this code  here https://github.com/wordpress-mobile/WordPress-Android/pull/9544/files#diff-38b6aea9cfff5f3e0e4b7389b44de602R2020

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java [hunk 2]
```diff
@@ -1972,6 +1970,10 @@ private void saveResult(boolean saved, boolean discardable, boolean savedLocally
     }
 
     private void publishPost() {
+        publishPost(false);
+    }
+
+    private void publishPost(final boolean isDraftToPublish) {
```
Comment (hypest) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: 👋 @mzorz , I wonder if we can do the `PostStatus.fromPost(mPost) == PostStatus.DRAFT` check inside the method and we go back to only have one parameter-less `publish()` method.

It looks like the only time we call this method directly with a true/false literal is [from showPublishConfirmationOrUpdateIfNotLocalDraft](https://github.com/wordpress-mobile/WordPress-Android/blob/e06af40f34a4772f7376baae5b1f6f87f5360e3d/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java#L1495) and it seems we can afford to redo the `PostStatus.fromPost(mPost) == PostStatus.DRAFT` in that case? WDYT?

File: WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadUtils.java [hunk 3]
```diff
@@ -252,8 +252,7 @@ public static void publishPost(Activity activity, final PostModel post, SiteMode
         }
 
         PostUtils.updatePublishDateIfShouldBePublishedImmediately(post);
-        boolean isFirstTimePublish = PostStatus.fromPost(post) == PostStatus.DRAFT
-                                     || (PostStatus.fromPost(post) == PostStatus.PUBLISHED && post.isLocalDraft());
+        boolean isFirstTimePublish = PostUtils.isFirstTimePublish(post);
```
Comment (hypest) [inline WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadUtils.java]: Good catch 👍

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java (hunk 2 above)
Comment (mzorz) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: Thank you! Let's adress separately as agreed
